src/components/AddNote.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useContext } from 'react';
+import styled from 'styled-components';
+import { Icon } from 'semantic-ui-react';
+import { mainStateDispatchContext } from './Main';
+
+const AddNoteWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  color: black;
+`;
+
+const AddNoteButton = styled.button`
+  background: none;
+  border: none;
+  outline: none;
+  font-size: 3rem;
+  color:inherit;
+`;
+
+function AddNote() {
+  /**
+   * the dispatch function of the state of 'Main' component
+   * @param {object} action
+   */
+  const mainStateDispatchFunction = useContext(mainStateDispatchContext);
+
+  const [noteTemp, setNoteTemp] = useState({
+    id: '',
+    body: '',
+    color: '#FEF3BD',
+    timeAdded: '',
+  });
+
+  const addIcon = <Icon name="add" />;
+
+  /**
+   * attaches a unique id to a blank note defined in its state and sends to
+   * Main component's dispatch function to
+   */
+  const onAdd = () => {
+    const dateAdded = new Date();
+    const addTime = dateAdded.getTime();
+    mainStateDispatchFunction({ type: 'add', state: noteTemp, timeAdded: addTime });
+    // the code below is to generate a new id for the next onAdd() event
+    const uniqueId = Math.floor(Math.random() * (10 ** 9) + 1);
```

Comment:
Here is the problem.

This id MAY not be unique.  use  `new Date().getUTCMilliseconds();` Still it may NOT be unique but it is much better than what you have.

Best solution: OUT of this function declare let id = 0; and increment it INSIDE this function.
==> any time this function is called it increments and the IDs are unique.

Reply:
Yes, but if the page weren't static and kept the notes on the server, it wouldn't be unique, would it? I changed it to the first approach ( new Date().getUTCMilliseconds() ) 🙂